src/commands/createHttpFunction.ts

```diff
@@ -65,3 +62,29 @@ function generateSuffixedName(preferredName: string, i: number): string {
 async function isNameAvailable(folderPath: string, newName: string): Promise<boolean> {
     return !(await fse.pathExists(path.join(folderPath, newName)));
 }
+
+async function getFunctionsApi(context: IActionContext): Promise<AzureFunctionsExtensionApi> {
+    const funcExtensionId: string = 'ms-azuretools.vscode-azurefunctions';
+    const funcExtension: vscode.Extension<AzureExtensionApiProvider> | undefined = vscode.extensions.getExtension(funcExtensionId);
+
+    if (funcExtension) {
+        if (!funcExtension.isActive) {
+            await funcExtension.activate();
+        }
+
+        // The Functions DB extension just recently added support for 'AzureExtensionApiProvider' so we should do an additional check just to makes sure it's defined
+        // tslint:disable-next-line: strict-boolean-expressions
+        if (funcExtension.exports) {
+            return funcExtension.exports.getApi<AzureFunctionsExtensionApi>('^1.0.0');
+        }
+    }
+
+    await ext.ui.showWarningMessage(localize('funcInstall', 'You must have the "Azure Functions" extension installed to perform this operation.'), { title: 'Install' });
+    const commandToRun: string = 'extension.open';
+    vscode.commands.executeCommand(commandToRun, funcExtensionId);
+
+    context.errorHandling.suppressDisplay = true;
+    context.telemetry.properties.installFunctions = 'true';
+    // we still need to throw an error even if the user installs
```

Comment:
Use `cancelStep` so that this can show up in existing queries, otherwise we might never write a query to check this new property `installFunctions`
```
context.telemetry.properties.cancelStep = 'installFunctions';
throw new UserCancelledError();
```

Also you don't need `suppressDisplay` if you treat this as a cancel.